pkg/spec/parse.go

```diff
@@ -138,22 +138,25 @@ func validateIOpsDevice(val string) (*throttleDevice, error) { //nolint
 }
 
 // getLoggingOpts splits the path= and tag= options provided to --log-opt.
-func getLoggingOpts(opts []string) (string, string) {
+func getLoggingOpts(opts []string) (string, string, string) {
 	var path, tag string
+	var logsizemax string
 	for _, opt := range opts {
-		arr := strings.SplitN(opt, "=", 2)
-		if len(arr) == 2 {
+		arr := strings.SplitN(opt, "=", 4)
```

Comment:
I think you want 3 here
```suggestion
		arr := strings.SplitN(opt, "=", 3)
```

Reply:
as per @rhatdan valuable suggestions introduced map return from getLoggingOpts function now it can handle any number of --log-opt <A=X> <B=Y> .... options supplied by --log-opt option field. Tested my code it is working fine.

Changes are as the below

func getLoggingOpts(opts []string) map[string]string {
        var logMap map[string]string
        for _, opt := range opts {
                arr := strings.SplitN(opt, "=", 2) /* for any --log-opt A=X B=Y C=D... it should loop over & split into "A" "X" fields 
                if len(arr) == 2 {
                        if strings.TrimSpace(arr[0]) == "path" {
                                logMap["path"] = strings.TrimSpace(arr[1])
                        }
                        if strings.TrimSpace(arr[0]) == "tag" {
                                logMap["tag"] = strings.TrimSpace(arr[1])
                        }
                        if strings.TrimSpace(arr[0]) == "max-size" {
                                logMap["logsizemax"] = strings.TrimSpace(arr[1])
                        }
                }
                if logMap["path"] != "" && logMap["tag"] != "" && logMap["logsizemax"] != "" {
                        break
                }
        }
        return logMap
}

Kindly further review. Thanks